app/controllers/concerns/phone_confirmation_flow.rb

```diff
@@ -84,4 +110,18 @@ def clear_session_data
     user_session.delete(unconfirmed_phone_session_key)
     user_session.delete(confirmation_code_session_key)
   end
+
+  def sms_enabled?
```

Comment:
For now, I will review this PR and agree to merge it as is, and then I will start refactoring these concerns.